src/AppBundle/Manager/UserManager.php

```diff
@@ -327,6 +327,30 @@ public function getCategoriesExactMatchByUser(User $user, User $currentUser)
             $categories     = implode(', ', $categories);
         }
         return $categories;
+    }
+	/**
+     * @param User $user
+     * @param User $currentUser
+	 * @param $locale
+     * @return string
+     */
+    public function getCategoriesExactMatchByUserAndLocale(User $user, User $currentUser, $locale)
+    {
+        $categories             = [];
+        $currentUserCategories  = $this->categoryManager->getFindByIdsAndLocale(array_keys($currentUser->getCategoryNames()), $locale);
+        $userCategories         = $user->getCategoryNames();
+        foreach($currentUserCategories as $currentUserCategory) {
+            if (isset($userCategories[$currentUserCategory->getId()])) {
+                $categories[]   = $currentUserCategory->getName();
+            }
+        }
+        if (count($categories) > 1) {
+            $lastCategory   = array_pop($categories);
+            $categories     = implode(', ', $categories) .' '.  ($locale == 'en'? 'and':'och') .' '. $lastCategory;
```

Comment:
Use the `translator`-service and pass $locale as the fourth parameter. Or even better if you can refactor this common code to a separate service.